### PR TITLE
Move pieces

### LINKED ISSUE #8: Implement possibility to move any chess piece to any empty square
Seems like a simple logic to be implemented

### PR DESCRIPTION
**Pull request by khaletska**
- Added possibility to move any piece to any empty square
- Replaced all the cell coordinates occurrences with Coordinate structure

### CODE REVIEW

File: chess/ChessGameViewModel.swift [hunk 1]
```diff
@@ -9,41 +9,45 @@ import SwiftUI
...
     }
 
-    func selectPieceAt(row: Int, col: Int) {
-        guard let selectedPieceAddress = self.selectedPieceAddress else {
-            self.selectedPieceAddress = (row, col)
-            return
-        }
-
-        if selectedPieceAddress == (row, col) {
+    func handleTap(at cellAddress: Coordinate) {
```
Comment (khaletska) [inline chess/ChessGameViewModel.swift]: function naming is up for discussion

Comment (Shiaulis) [reply chess/ChessGameViewModel.swift]: cell Tapped?

File: chess/ChessGameViewModel.swift [hunk 2]
```diff
@@ -9,41 +9,45 @@ import SwiftUI
...
+            self.selectedPieceAddress = cellAddress
+        case (true, true):
+            print("empty cell tapped and we don't have selected piece")
+            return
         }
     }
 
-    func getPieceForCell(row: Int, col: Int) -> ChessPiece? {
-        self.model.board[row][col]
+    func getPiece(for cell: Coordinate) -> ChessPiece? {
```
Comment (Shiaulis) [inline chess/ChessGameViewModel.swift]: let's unify naming for parameters that have `Coordinate` type. Here it is a `cell` but down below it is `address`. I assume word `cell` is recerved by the logic so coordinate(s) or address would make more sense

**Review by Shiaulis — CHANGES_REQUESTED** [review]
works good, let's just improve naming a bit

**Review by Shiaulis — APPROVED** [review]

**Pull request merged**